docs/current_docs/api/secrets.mdx

```diff
@@ -60,6 +60,8 @@ dagger core container \
 
 [Secret arguments](./arguments.mdx#secret-arguments) can be sourced from multiple providers: the host environment, the host filesystem, the result of host command execution, and external secret managers [1Password](https://1password.com/) and [Vault](https://www.hashicorp.com/products/vault).
 
+When a `Secret` is included in other operations, the cache for those operations will be based on the plaintext value of the secret. If the same operation is run with a secret with the same plaintext value, that operation may be cached rather than re-executed. In the above example, the secret is specified as `env://API_TOKEN`, so the cache for the `with-exec` will be based on the plaintext value of the secret. If two clients execute the container and have the same `API_TOKEN` environment variable value, they may share cache for the container. However, if two clients have different values for the `API_TOKEN` environment variable, the `with-exec` will not share cache, even though both clients specify the secret as `env://API_TOKEN`.
+
 ## Security considerations
```

Comment:
Are there any security considerations to be aware of around secret caching? If yes, they could be added here.

Reply:
Not really anything the user needs to be aware of or take action from. There's some implementation details [mentioned under "Security" in this PR description](https://github.com/dagger/dagger/pull/10311) but based on what we have in the docs today I don't think it's anything we need to mention here.